spot

```diff
@@ -1,64 +1,59 @@
 #!/bin/bash
-if [[ $# != 1 ]]
-then
-	printf "Usage\n\tspot <URL>\n"
-	printf "\nExample\n\thttp://open.spotify.com/user/1248318865/playlist/7uKPSEwUQzx3c9EU032lU6\n"
-	exit 0
-fi
 
-input=$1
-keyword=spotify
-schemedurl=
-url=
+# enable shell script debugging
+# set -x
 
-matches() {
-    input="$1"
-    pattern="$2"
-    echo "$input" | grep -q "$pattern"
-}
-
-if matches $input $keyword
+# display help if incorrect usage or on --help
+if [[ $1 == "--help" ]]
 then
-        url=$input
-else
-        echo "$1 is not a valid spotify url" 
-	exit 0;
+	  printf "Usage\n\tspot <spotify web url OR search string>\n\tspot --help to see this help text\n"
+	  printf "\nExamples\n\tspot http://open.spotify.com/user/1248318865/playlist/7uKPSEwUQzx3c9EU032lU6\n"
+    printf "\tspot breaking benjamin\n"
```

Comment:
Gah sorry about this. I use white spaces for alignment and you used hard tabs. What would you prefer I use?